aiosmtpd/smtp.py

```diff
@@ -438,7 +438,12 @@ def smtp_DATA(self, arg):
                 'mail_options': self.mail_options,
                 'rcpt_options': self.rcpt_options,
             }
-        status = self.event_handler.process_message(*args, **kwargs)
+        if asyncio.iscoroutinefunction(self.event_handler.process_message):
+            kwargs['loop'] = self.loop
+            status = yield from self.event_handler.process_message(
+                *args, **kwargs)
+        else:
+            status = self.event_handler.process_message(*args, **kwargs)
```

Comment:
I wish there was a way to do the same call in all cases.  Can you think of a way to do that?  It's another place where I wish we didn't have to do a typecheck on the method.  If not, oh well.